cli/args/mod.rs

```diff
@@ -1057,70 +1058,78 @@ impl CliOptions {
-        base_url: root_config_file.specifier.clone(),
-        import_map_value: root_config_file.to_import_map_value_from_imports(),
-      };
-      let children_configs = workspace_config
-        .members
-        .iter()
-        .map(|member| ::import_map::ext::ImportMapConfig {
-          base_url: member.config_file.specifier.clone(),
-          import_map_value: member
-            .config_file
-            .to_import_map_value_from_imports(),
-        })
-        .collect();
-
-      let (import_map_url, import_map) =
-        ::import_map::ext::create_synthetic_import_map(
-          base_import_map_config,
-          children_configs,
-        );
-      let import_map = enhance_import_map_value_with_workspace_members(
-        import_map,
-        &workspace_config.members,
-      );
-      log::debug!(
-        "Workspace config generated this import map {}",
-        serde_json::to_string_pretty(&import_map).unwrap()
-      );
-      let maybe_import_map_result =
-        import_map::import_map_from_value(import_map_url, import_map).map(Some);
-
-      return maybe_import_map_result;
-    }
-
-    if self
+  ) -> Result<WorkspaceResolver, AnyError> {
+    let overrode_no_import_map = self
       .overrides
       .import_map_specifier
       .as_ref()
       .map(|s| s.is_none())
-      == Some(true)
-    {
-      // overrode to not use an import map
-      return Ok(None);
-    }
-
-    let import_map_specifier = self.resolve_specified_import_map_specifier()?;
-    resolve_import_map(
-      import_map_specifier.as_ref(),
-      self.maybe_config_file().as_ref(),
-      file_fetcher,
+      == Some(true);
+    let cli_arg_specified_import_map = if overrode_no_import_map {
+      // use a fake empty import map
+      Some(deno_config::workspace::SpecifiedImportMap {
+        base_url: self
+          .workspace
+          .root_folder()
+          .0
+          .join("import_map.json")
+          .unwrap(),
+        value: serde_json::Value::Object(Default::default()),
+      })
+    } else {
+      let maybe_import_map_specifier =
+        self.resolve_specified_import_map_specifier()?;
+      match maybe_import_map_specifier {
+        Some(specifier) => {
+          let value =
+            resolve_import_map_value_from_specifier(&specifier, file_fetcher)
+              .await
+              .with_context(|| {
+                format!("Unable to load '{}' import map", specifier)
+              })?;
+          Some(deno_config::workspace::SpecifiedImportMap {
+            base_url: specifier,
+            value,
+          })
+        }
+        None => None,
+      }
+    };
+    Ok(
+      self
+        .workspace
+        .create_resolver(
+          CreateResolverOptions {
+            // todo(dsherret): this should be false for nodeModulesDir: true
+            pkg_json_dep_resolution: if self.use_byonm() {
+              PackageJsonDepResolution::Disabled
```

Comment:
Do you want to address that before landing?

Reply:
No, that's a big change.